src/test/common/test_random.cc

```diff
@@ -0,0 +1,224 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2017 SUSE LINUX GmbH
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+*/
+
+#include <sstream>
+
+#include "include/random.h"
+
+#include "gtest/gtest.h"
+
+// Helper to see if calls compile with various types:
+template <typename T>
+T type_check_ok(const T min, const T max)
+{
+  return ceph::util::generate_random_number(min, max);
+}
+
+/* This mixes integer and real types freely, so it's not the greatest
+of tests, but should work casually for us: */
+template <typename X0, typename X1>
+bool none_equal(const X0 x0, const X1 x1)
+{
+  return x0 != x1;
+}
+
+template <typename X0, typename X1, typename ...XS>
+bool none_equal(const X0 x0, const X1 x1, const XS ...xs)
+{
+  return none_equal(x0, none_equal(x1, xs...));
```

Comment:
this doesn't look right? it compares `x0` with the boolean result of `none_equal()`?